shell/machine-config/pnap.vue

```diff
@@ -0,0 +1,279 @@
+      osChoices:       null,
+    };
+  },
+
+  watch: {
+    'credentialId'() {
+      this.$fetch();
+    },
+  },
+
+  computed: {
+    isCreate() {
+      return this.mode === _CREATE;
+    },
+    locationChoices() {
+      const products = this.allProducts;
+      const result = [];
+
+      if (Array.isArray(products)) {
+        const map = new Map();
+
+        for (const prod of products) {
+          for (const plan of prod.plans) {
+            if (!map.has(plan.location)) {
+              map.set(plan.location, true);
+              result.push({ value: plan.location, label: plan.location });
+            }
+          }
+        }
+      } else {
+        result.push({ value: products.message });
+      }
+
+      return result;
+    },
+    typeChoices() {
+      const products = this.allProducts;
+      const result = [];
+
+      if (Array.isArray(products)) {
+        const map = new Map();
+        const serverLocation = this.value.serverLocation;
+
+        for (const prod of products) {
+          for (const plan of prod.plans) {
+            if (
+              !map.has(prod.productCode) &&
+              plan.location === serverLocation
+            ) {
+              map.set(prod.productCode, true);
+              result.push({
+                value: prod.productCode,
+                label: `${ prod.productCode } - $${ plan.price }/hour`,
+              });
+            }
+          }
+        }
+        if (!map.has(this.value.serverType) && products.length > 0) {
+          this.preselectFirst(result[0].value);
+        }
+      } else {
+        result.push({ value: products.message });
+      }
+
+      return result;
+    },
+    type() {
+      const products = this.allProducts;
+
+      if (Array.isArray(products)) {
+        const serverLocation = this.value.serverLocation;
+        const serverType = this.value.serverType;
+
+        for (const prod of products) {
+          for (const plan of prod.plans) {
+            if (prod.productCode === serverType && plan.location === serverLocation) {
+              return prod;
+            }
+          }
+        }
+      }
+
+      return products.message;
+    },
+  },
+
+  methods: {
+    stringify,
+
+    updateInterfaces() {
+      if (!this.value.interfaces) {
+        this.value.interfaces = [];
+      }
+    },
+
+    preselectFirst(val) {
+      this.value.serverType = val;
+    },
+  },
+};
+</script>
+
+<template>
+  <Loading v-if="$fetchState.pending" :delayed="true" />
+  <div v-else-if="errors.length">
+    <div v-for="(err, idx) in errors" :key="idx">
+      <Banner color="error" :label="stringify(err)" />
+    </div>
+  </div>
+  <div v-else>
+    <div class="row mt-20">
+      <div class="col span-6">
+        <LabeledSelect
+          v-model="value.serverLocation"
+          :mode="mode"
+          :options="locationChoices"
+          :searchable="true"
+          :required="true"
+          :disabled="disabled"
+          label="Location"
```

Comment:
These need localising. I've added it to this line and the pattern should reproducible

Reply:
These have been localized.